packages/base/src/processing/processingCommands.ts

```diff
@@ -0,0 +1,85 @@
+import {
+  IDict,
+  IJGISFormSchemaRegistry,
+  ProcessingMerge,
+  ProcessingLogicType,
+  ProcessingType,
+} from '@jupytergis/schema';
+import { JupyterFrontEnd } from '@jupyterlab/application';
+import { CommandRegistry } from '@lumino/commands';
+
+import { selectedLayerIsOfType, processSelectedLayer } from './index';
+import { JupyterGISTracker } from '../types';
+
+export function replaceInSql(
+  sql: string,
+  keyToVal: IDict<string>,
+  layerName: string,
+) {
+  const replaceTemplateString = (args: {
+    variableName: string;
+    template: string;
+    value: string;
+  }): string =>
+    args.template.replace(RegExp(`{${args.variableName}}`, 'g'), args.value);
+
+  let out = replaceTemplateString({
+    variableName: 'layerName',
+    template: sql,
+    value: layerName,
+  });
+
+  for (const [key, value] of Object.entries(keyToVal)) {
+    out = replaceTemplateString({
+      variableName: key,
+      template: out,
+      value: value,
+    });
+  }
+
+  return out;
+}
+
+export function addProcessingCommands(
+  app: JupyterFrontEnd,
+  commands: CommandRegistry,
+  tracker: JupyterGISTracker,
+  trans: any,
+  formSchemaRegistry: IJGISFormSchemaRegistry,
+) {
+  for (const processingElement of ProcessingMerge) {
+    if (processingElement.processType === ProcessingLogicType.vector) {
```

Comment:
Can we add an `else` condition here that raises an error for any `ProcessType` with undefined behavior?